sdk/servicebus/azure-servicebus/azure/servicebus/aio/_base_handler_async.py

```diff
@@ -147,42 +148,51 @@ async def _backoff(
             raise last_exception
 
     async def _do_retryable_operation(self, operation, timeout=None, **kwargs):
+        # type: (Callable, Optional[float], Any) -> Any
         require_last_exception = kwargs.pop("require_last_exception", False)
         require_timeout = kwargs.pop("require_timeout", False)
         retried_times = 0
         last_exception = None
         max_retries = self._config.retry_total
 
+        abs_timeout_time = (time.time() + timeout) if (require_timeout and timeout) else None
+
         while retried_times <= max_retries:
             try:
-                if require_last_exception:
-                    kwargs["last_exception"] = last_exception
-                if require_timeout:
-                    kwargs["timeout"] = timeout
+                if require_timeout and abs_timeout_time:
```

Comment:
Nit: I realize the naming is likely to be consistent, but the purpose is different (require_timeout, require_last_exception); where the former is more of "operation_requires_timeout" and the latter "should_raise_last_exception"  (Being verbose for making a point; take it or leave it or mutate as you see fit, this is a very bikesheddy change so mostly using it as a chance to wax poetic)

Reply:
good suggestion, renamed `require_timeout` to `operation_requires_timeout` to make it clear.
keeping `last_exception` as it holds exception, not a flag